src/targets/shell/curl.js

```diff
@@ -74,10 +75,38 @@ module.exports = function (source, options) {
     default:
       // raw request body
       if (source.postData.text) {
-        code.push(
-          '%s %s', opts.binary ? '--data-binary' : (opts.short ? '-d' : '--data'),
-          helpers.quote(source.postData.text)
-        )
+        let builtPayload = false
+
+        // If we're dealing with a JSON variant, and our payload is JSON let's make it look a little nicer.
+        if (headerHelpers.isMimeTypeJson(source.postData.mimeType)) {
+          // If our postData is less than 20 characters, let's keep it all on one line so as to not make the snippet
+          // overly lengthy
+          if (source.postData.text.length > 20) {
```

Comment:
Mind adding a test for this short case?

Reply:
It's already covered with the `jsonObj-null-value` case, but I'll add an explicit test for it.